test/integration/master_test_utils.go

```diff
@@ -0,0 +1,198 @@
+/*
+Copyright 2014 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/controller"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/controller/framework"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/master"
+	"github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/admission/admit"
+	"github.com/golang/glog"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"time"
+)
+
+// Timeout used in benchmarks, to eg: resize an rc
+var Timeout = 30 * time.Minute
+
+// TODO: Include all master components (scheduler, nodecontroller).
+// TODO: Reconcile with integration.go, currently the master used there doesn't understand
+// how to restart cleanly, which is required for each iteration of a benchmark. The integration
+// tests also don't make it easy to isolate and turn off components at will.
+type MasterComponents struct {
+	// Raw http server in front of the master
+	apiServer *httptest.Server
+	// Kubernetes master, contains an embedded etcd helper
+	kubeMaster *master.Master
+	// Restclient used to talk to the kubernetes master
+	restClient *client.Client
+	// Replication controller manager
+	controllerManager *controller.ReplicationManager
+	// Channel for stop signals to all master components
+	stopCh chan struct{}
+}
+
+// TODO: Run these in containers each with their own --parent-cgroup so they don't step on each other?
+func NewMasterComponents() *MasterComponents {
```

Comment:
Sure, i was going for the opposite, I'll give you everything and you stop what you want, mostly because I sometimes find it a pain to write a test if I need to create a config that required 10 other settings I don't understand. configurability is also important though, what are your thoughts on this tradeoff?